cmd/archeio/internal/app/buckets.go

```diff
@@ -91,6 +92,15 @@ func newCachedBlobChecker() *cachedBlobChecker {
 		blobCache: blobCache{
 			cache: make(map[string]map[string]struct{}),
 		},
+		client: newHTTPClient(),
+	}
+}
+
+func newHTTPClient() *http.Client {
+	// ensure sensible timeouts
+	// this client will be used to make HEAD calls
+	return &http.Client{
```

Comment:
TODO: transport and connection reuse tuning, given the large amount of calls to same-host?

Reply:
we do, and that will implicitly use http.DefaultTransport
http client transport can be tuned for connection idle + reuse

Reply:
client reuse may actually be suboptimal (locks on shared resources like cookies etc) vs just ensuring transport reuse and tuning that. 

Most advice about client re-use seems to be either related to transport reuse (which can instead be accomplished by passing a transport when constructing a client) or maybe allocation (not a primary concern here).